engine/repositories/processor_push.go

```diff
@@ -24,29 +34,28 @@ func (s *Service) processPush(ctx context.Context, op *sdk.Operation) error {
 		op.Setup.Push.ToBranch = op.RepositoryInfo.DefaultBranch
 	}
 
-	// Switch to default branch
-	if currentBranch != op.RepositoryInfo.DefaultBranch {
-		if err := gitRepo.FetchRemoteBranch("origin", op.RepositoryInfo.DefaultBranch); err != nil {
-			log.Error(ctx, "Repositories> processPush> Checkout to default branch> [%s] error %v", op.UUID, err)
-			return sdk.WrapError(err, "Checkout to default branch> [%s] error %v", op.UUID, err)
+	// Switch to target branch
+	if currentBranch != op.Setup.Push.FromBranch {
+		if err := gitRepo.CheckoutNewBranch(op.Setup.Push.FromBranch); err != nil {
```

Comment:
is it not better to go to default branch, git pull and then create the new branch?

Reply:
I changed the way it works to remove usage of FetchRemoteBranch func from go-repo. Now there is a hars reset to origin/defaultBranch if the branch already exists locally but was never pushed.